database.go

```diff
@@ -0,0 +1,1021 @@
+	RestoreFromBackup(ctx context.Context, databaseID string, databaseRestoreReq *DatabaseBackupRestoreReq) (*Database, error)
+	Fork(ctx context.Context, databaseID string, databaseForkReq *DatabaseForkReq) (*Database, error)
+
+	ListConnectionPools(ctx context.Context, databaseID string) (*DatabaseConnections, []DatabaseConnectionPool, *Meta, error)
+	CreateConnectionPool(ctx context.Context, databaseID string, databaseConnectionPoolReq *DatabaseConnectionPoolCreateReq) (*DatabaseConnectionPool, error)
+	GetConnectionPool(ctx context.Context, databaseID string, poolName string) (*DatabaseConnectionPool, error)
+	UpdateConnectionPool(ctx context.Context, databaseID string, poolName string, databaseConnectionPoolReq *DatabaseConnectionPoolUpdateReq) (*DatabaseConnectionPool, error)
+	DeleteConnectionPool(ctx context.Context, databaseID string, poolName string) error
+
+	ListAdvancedOptions(ctx context.Context, databaseID string) (*DatabaseAdvancedOptions, []AvailableOption, error)
+	UpdateAdvancedOptions(ctx context.Context, databaseID string, databaseAdvancedOptionsReq *DatabaseAdvancedOptions) (*DatabaseAdvancedOptions, []AvailableOption, error)
+
+	ListAvailableVersions(ctx context.Context, databaseID string) ([]string, error)
+	StartVersionUpgrade(ctx context.Context, databaseID string, databaseVersionUpgradeReq *DatabaseVersionUpgradeReq) (string, error)
+}
+
+// DatabaseServiceHandler handles interaction with the server methods for the Vultr API
+type DatabaseServiceHandler struct {
+	client *Client
+}
+
+type DBPlanListOptions struct {
+	Engine string `url:"engine,omitempty"`
+	Nodes  int    `url:"nodes,omitempty"`
+	Region string `url:"region,omitempty"`
+}
+
+// DatabasePlan represents a Managed Database plan
+type DatabasePlan struct {
+	ID               string           `json:"id"`
+	NumberOfNodes    int              `json:"number_of_nodes"`
+	Type             string           `json:"type"`
+	VCPUCount        int              `json:"vcpu_count"`
+	RAM              int              `json:"ram"`
+	Disk             int              `json:"disk"`
+	MonthlyCost      int              `json:"monthly_cost"`
+	SupportedEngines SupportedEngines `json:"supported_engines"`
+	MaxConnections   *MaxConnections  `json:"max_connections,omitempty"`
+	Locations        []string         `json:"locations"`
+}
+
+// SupportedEngines represents an object containing supported database engine types for Managed Database plans
+type SupportedEngines struct {
+	MySQL *bool `json:"mysql"`
+	PG    *bool `json:"pg"`
+	Redis *bool `json:"redis"`
+}
+
+// MaxConnections represents an object containing the maximum number of connections by engine type for Managed Database plans
+type MaxConnections struct {
+	MySQL int `json:"mysql,omitempty"`
+	PG    int `json:"pg,omitempty"`
+}
+
+type databasePlansBase struct {
+	DatabasePlans []DatabasePlan `json:"plans"`
+	Meta          *Meta          `json:"meta"`
+}
+
+type DBListOptions struct {
```

Comment:
exported type DBPlanListOptions should have comment or be unexported

Reply:
added the missing comments